### PR TITLE
add --quiet feature

### PR DESCRIPTION
**Pull request by beatfreaker**
Fix for https://github.com/sindresorhus/xo/issues/125

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you add a test?

**Comment by beatfreaker** [comment]
yup, doing that

and one more question does have to implement the same code in `lintText` function as well?

**Comment by sindresorhus** [comment]
> and one more question does have to implement the same code in lintText function as well?

Yes

File: index.js [hunk 1]
```diff
@@ -64,7 +65,9 @@ function mergeReports(reports) {
 function runEslint(paths, opts) {
 	var config = optionsManager.buildConfig(opts);
 	var engine = new eslint.CLIEngine(config);
-	return engine.executeOnFiles(paths, config);
+	var report = engine.executeOnFiles(paths, config);
+	report.results = opts.quiet ? eslint.CLIEngine.getErrorResults(report.results) : report.results;
```
Comment (jamestalmage) [inline index.js]: Add a new `processReport(report, opts)` function that contains this common code:

```js
function processReport(report, opts) {
  report.results = ....
  return report;
}

**Comment by jamestalmage** [comment]
LGTM

File: test/cli.js [hunk 2]
```diff
@@ -37,3 +37,13 @@ test('supports being extended with a shareable config', async () => {
 	const cwd = path.join(__dirname, 'fixtures/project');
 	await execa('../../../cli.js', ['--no-local'], {cwd});
 });
+
+test('quiet option', async t => {
+	const filepath = await tempWrite('// TODO: quiet\nconsole.log()\n', 'x.js');
+
+	try {
+		await execa('../cli.js', ['--no-local', '--quiet', '--reporter=compact', filepath]);
```
Comment (sindresorhus) [inline test/cli.js]: You can use `t.throws` here. It returns the error which you then can check.

https://github.com/avajs/ava#throwsfunctionpromise-error-message

File: test/cli.js [hunk 3]
```diff
@@ -37,3 +37,13 @@ test('supports being extended with a shareable config', async () => {
...
+	} catch (err) {
+		t.true(err.stdout.indexOf('warning') === -1);
```
Comment (sindresorhus) [inline test/cli.js]:
```js
t.is(err.stdout.indexOf('warning'),-1);
```

File: test/cli.js [hunk 4]
```diff
@@ -37,3 +37,9 @@ test('supports being extended with a shareable config', async () => {
 	const cwd = path.join(__dirname, 'fixtures/project');
 	await execa('../../../cli.js', ['--no-local'], {cwd});
 });
+
+test('quiet option', async t => {
+	const filepath = await tempWrite('// TODO: quiet\nconsole.log()\n', 'x.js');
+	var err = await t.throws(execa('../cli.js', ['--no-local', '--quiet', '--reporter=compact', filepath]));
```
Comment (sindresorhus) [inline test/cli.js]: `var` => `const`

**Comment by sindresorhus** [comment]
👍 Awesome :)

**Pull request merged**